### ISSUE TITLE
Schema file MTConnectStreams_2.0.0_1.0.xsd is not valid

### ISSUE THREAD

**Issue by hulinning2** [mtconnect/cppagent#230]
Using MTConnect Agent and schema files from version 2.0.0.6

I got this error from Agent MTConnectStreams
![image](https://user-images.githubusercontent.com/4930742/171666372-c11810be-a02d-4424-b7fd-5d300a289d01.png)

I checked the MTConnectStreams_2.0.0_1.0.xsd file itself and has these errors:
![image](https://user-images.githubusercontent.com/4930742/171665653-b198b794-0d01-4328-ba6d-df86def8e4a3.png)

It seems I got the same issue with spec 1.7 and 1.8 also

Thanks

**Comment by wsobel** [comment]
I'll check it out. I thought we had tested these schemas.

**Comment by wsobel** [comment]
This is related to the new 1.0 generator. We need to update the generator for the various versions.

**Comment by wsobel** [comment]
Assigning to Shaurabh

**Comment by wsobel** [comment]
The ``xsd:all` schema tag does not support unlimited, that was a 1.1 addition. We need to change to `xsd:sequence`.

From:
```xml
    <xs:all>
      <xs:element ref='Sample' minOccurs='1' maxOccurs='unbounded'/>
    </xs:all>
```

To:
```xml
    <xs:sequence>
      <xs:element ref='Sample' minOccurs='1' maxOccurs='unbounded'/>
    </xs:sequence>
```

**Comment by hulinning2** [comment]
Still got the same error in release 2.0.0.7 due to this

**Comment by wsobel** [comment]
I will check. I am putting together another build and the files may not have been copied from schema repo. Check schemas.mtconnect.org. If it is still an issue there, then we can fix. 

If not, then we will patch once I release 8. 

(Sent from mobile)

**Comment by hulinning2** [comment]
just checked from schemas.mtconnect.org and got the same issue

**Comment by wsobel** [comment]
Thanks Linh. I think Shaurabh just fixed the schema repo. I’ll review and merge soon to the schemas. If that fixes the issue, I’ll start putting together a .8 release.

Thx,
W

**Comment by wsobel** [comment]
Merged in schemas. Please re-test and tell me if it is fixed.

**Comment by hulinning2** [comment]
Fixed in 2.0 but 1.8 and 1.7 still has the same issue.
I am working on spec 1.7 now.
Thanks

**Comment by shaurabhsingh** [comment]
Fixed now in 1.7 and 1.8 as well. Please check the schema repository.

**Comment by hulinning2** [comment]
just checked 1.7 and 1.8 and are still working and same errors

**Comment by hulinning2** [comment]
MTConnect XML Schema Files<http://schemas.mtconnect.org/>

Linh Huynh
Senior Software Engineer
Okuma America Corporation
***@***.******@***.***>
704-504-6116
http://www.okuma.com/americas

VISIT OUR BOOTH AT IMTS

[IMTS Logo Horizontal with Dates]<https://www.okuma.com/imts>


Okuma America Corporation
11900 Westhall Drive
Charlotte, NC  28278


Visit or Follow us at:
Website<https://www.okuma.com/americas> | LinkedIn<http://www.linkedin.com/company/105029> | YouTube<https://www.youtube.com/user/OkumaAmericaCorp> | Instagram<https://www.instagram.com/okumaamerica/> | Facebook<https://www.facebook.com/OkumaAmericaCorporation/> | Twitter<https://twitter.com/OkumaAmerica> | Okuma App Store<https://www.myokuma.com/>

We passionately pursue a customer for life.

***@***.***<https://www.okuma.com/>

THIS COMMUNICATION IS INTENDED FOR THE SOLE USE OF THE PERSON TO WHOM IT IS ADDRESSED AND MAY CONTAIN INFORMATION THAT IS PRIVILEGED, CONFIDENTIAL AND EXEMPT FROM DISCLOSURE UNDER APPLICABLE LAW. ANY DISSEMINATION, DISTRIBUTION OR DUPLICATION OF THIS COMMUNICATION BY SOMEONE OTHER THAN THE INTENDED RECIPIENT IS STRICTLY PROHIBITED. IF YOUR RECEIPT OF THIS COMMUNICATION IS IN ERROR, PLEASE NOTIFY THE SENDER AND DELETE THIS COMMUNICATION.

**Comment by shaurabhsingh** [comment]
could you try again. It takes about 5 minutes to automatically update on mtconnect.org . I see the updated ones now and they validate.

**Comment by hulinning2** [comment]
checked again but still having issues with 1.7 and 1.8 on MTConnectStreams_1.7_1.0.xsd and MTConnectStreams_1.8_1.0.xsd

**Comment by shaurabhsingh** [comment]
It is working on my end. Could you screenshot the error? Or is it the same as before?

**Comment by hulinning2** [comment]
![image](https://user-images.githubusercontent.com/4930742/173885790-7a138d00-ffb9-4830-a5a4-7c8172304eea.png)

**Comment by hulinning2** [comment]
![image](https://user-images.githubusercontent.com/4930742/173885916-fedd5ca2-7df3-4f54-ba06-b1b55a6d8c50.png)

**Comment by hulinning2** [comment]
I open these files directly on the website http://schemas.mtconnect.org/ and it has the same issues. 
The xs:all are still there

**Comment by wsobel** [comment]
Checked the log and we have pulled:

Wed Jun 15 15:20:03 UTC 2022
From https://github.com/mtconnect/schema
 * branch            master     -> FETCH_HEAD
   d597241..1d50e2f  master     -> origin/master
Updating d597241..1d50e2f
Fast-forward
 MTConnectAssets_1.7_1.0.xsd  | 19197 +++++++++++++++++++--------------------
 MTConnectAssets_1.8_1.0.xsd  | 19994 +++++++++++++++++++++--------------------
 MTConnectDevices_1.7_1.0.xsd | 11358 +++++++++++------------
 MTConnectDevices_1.8_1.0.xsd | 11624 ++++++++++++------------
 MTConnectError_1.7_1.0.xsd   |  5332 +++++------
 MTConnectError_1.8_1.0.xsd   |  5482 +++++------
 MTConnectStreams_1.7_1.0.xsd | 15416 +++++++++++++++----------------
 MTConnectStreams_1.8_1.0.xsd | 16196 ++++++++++++++++-----------------
 8 files changed, 52391 insertions(+), 52208 deletions(-)

Not sure what’s going on.

Best,
W

**Comment by shaurabhsingh** [comment]
Linh, can you maybe try deleting and downloading it again. not all `xs:all` have been replaced. some will still be in there.

**Comment by hulinning2** [comment]
Shaurabhsingh, 
I am trying to work on MTConnect spec 1.7 now and the file named MTConnectStreams_1.7_1.0.xsd is not valid. 
I just get it from http://schemas.mtconnect.org/ 

It would be the same issue on 1.8

**Comment by hulinning2** [comment]
[Devices.zip](https://github.com/mtconnect/cppagent/files/8921728/Devices.zip)
Here is the devices.xml and its extended schema files

**Comment by shaurabhsingh** [comment]
So the device file seems to be validating after making the below changes:
1. changing "xsi:schemaLocation" to "urn:mtconnect.org:MTConnectDevices:1.7 http://schemas.mtconnect.org/schemas/MTConnectDevices_1.7.xsd" since i didnt have the mtconnect schema files you had locally referenced.
       You can change it back to your local reference.
2. added "deviceModelChangeTime" attribute to the Header.
3. added '<Agent id="agent1" uuid="abcdef" name="agent1"/>' element under "Devices" since it is required starting 1.7.

I have also attached the link xsd file, the latest MTConnectStreams schema for 1.7 ad 1.8 that have been updated online. The Okuma streams file validates now as well. Let me know if you are able to successfully replicate it.

[Devices_updated.zip](https://github.com/mtconnect/cppagent/files/8921943/Devices_updated.zip)
